### PR TITLE
populate pd_applications_status_logs table 

### PR DESCRIPTION
**Pull request by davidsbailey**
Finishes [ACQ-196](https://codedotorg.atlassian.net/browse/ACQ-196). Depends on https://github.com/code-dot-org/code-dot-org/pull/48823, which creates the table. also see design considerations discussed in this [doc](https://docs.google.com/document/d/1JYEJxgYimJEsmEH4mpZKyZ-uDs-0IjXLqIlrBkFZ2Rs/edit#).

This PR updates the existing cronjob `create_rollup_tables` to also populate `pd_applications_status_logs` based on the data in the `pd_applications` table. More specifically, it only looks at teacher applications, and looks the `status_log` field. one new row in `pd_applications_status_logs` is created for each `status_log` entry.

### why status_log

I chose `status_log` over `status_timestamp_change_log` because `status_log` has less junk in it. when I create a new teacher application in the UI, here is what I see in the database:
```
[development] dashboard > puts JSON.pretty_generate JSON.parse Pd::Application::ApplicationBase.last.status_timestamp_change_log
[
  {
    "time": "2022-10-25T21:44:12.855Z",
    "title": "confirmation_email"
  },
  {
    "time": "2022-10-25T21:44:12.890Z",
    "title": "principal_approval_email"
  },
  {
    "title": "unreviewed",
    "changing_user_id": 1,
    "changing_user_name": "DaveAll",
    "time": "2022-10-25T21:44:12.896Z"
  }
]
=> nil
[development] dashboard > puts JSON.pretty_generate Pd::Application::ApplicationBase.last.status_log
[
  {
    "status": "unreviewed",
    "at": "2022-10-25T21:44:12.723Z"
  }
]
=> nil
```

status_log also looks like it is updated reliably:
* https://github.com/code-dot-org/code-dot-org/blob/e27850b06aa758f1048b2009789f8e8acd2bff0d/dashboard/app/models/pd/application/teacher_application.rb#L87
* https://github.com/code-dot-org/code-dot-org/blob/e27850b06aa758f1048b2009789f8e8acd2bff0d/dashboard/app/models/pd/application/teacher_application.rb#L353-L355

## sample data

Here is what some of the resulting data looks like when run in the adhoc with cloned production db:
```
mysql> select count(*) from pd_applications_status_logs;
+----------+
| count(*) |
+----------+
|    49117 |
+----------+
1 row in set (0.01 sec)

mysql> select * from pd_applications_status_logs limit 20;
+----+-------------------+-------------------------------+---------------------+----------+
| id | pd_application_id | status                        | timestamp           | position |
+----+-------------------+-------------------------------+---------------------+----------+
|  1 |             17298 | unreviewed                    | 2018-10-02 19:35:50 |        0 |
|  2 |             17298 | pending                       | 2019-02-13 21:16:24 |        1 |
|  3 |             17298 | accepted_no_cost_registration | 2019-02-15 19:54:50 |        2 |
|  4 |             17299 | unreviewed                    | 2018-10-03 17:39:20 |        0 |
|  5 |             17299 | pending                       | 2018-11-08 17:40:35 |        1 |
|  6 |             17299 | accepted_no_cost_registration | 2018-11-13 16:27:56 |        2 |
|  7 |             17299 | withdrawn                     | 2019-05-04 10:57:50 |        3 |
|  8 |             17300 | unreviewed                    | 2018-10-04 10:59:59 |        0 |
|  9 |             17300 | pending                       | 2019-02-12 12:41:33 |        1 |
| 10 |             17300 | accepted_not_notified         | 2019-02-23 18:14:07 |        2 |
| 11 |             17300 | accepted_notified_by_partner  | 2019-02-23 18:23:29 |        3 |
| 12 |             17300 | registration_sent             | 2019-03-05 19:19:11 |        4 |
| 13 |             17301 | unreviewed                    | 2018-10-04 12:07:17 |        0 |
| 14 |             17301 | accepted_notified_by_partner  | 2019-01-23 15:54:55 |        1 |
| 15 |             17301 | withdrawn                     | 2019-02-11 21:19:19 |        2 |
| 16 |             17303 | unreviewed                    | 2018-10-04 12:51:46 |        0 |
| 17 |             17303 | pending                       | 2019-02-12 12:41:13 |        1 |
| 18 |             17303 | withdrawn                     | 2019-02-23 18:14:31 |        2 |
| 19 |             17307 | unreviewed                    | 2018-10-04 18:53:54 |        0 |
| 20 |             17307 | accepted_not_notified         | 2019-01-17 14:59:44 |        1 |
+----+-------------------+-------------------------------+---------------------+----------+
20 rows in set (0.00 sec)
```

## Testing story

- [x] run `create_rollup_tables` locally on 1 pd application
- [x] run `create_rollup_tables` on an adhoc with cloned production database

the adhoc is named `adhoc-rollup-application-status-log` in case you want to poke around. It may need to be restarted first.

If for any reason we're not 100% confident that the output date looks right, we can always ship this PR and then let the data team experiment with the resulting data to see if it meets their needs, rather than tinkering around on the adhoc. The testing on the adhoc was mostly to make sure I didn't break the cronjob.

## Deployment strategy

no real reason to wait for the migration PR to ship ahead of this PR, because the cronjob only runs [once a week](https://github.com/code-dot-org/code-dot-org/blob/1cacc21f7ed1c308727c74f77197b6e692d583b2/cookbooks/cdo-apps/templates/default/crontab.erb#L130).

## Scalability

Because there will be more entries in the new table than the other tables populated by `create_rollup_tables` (or may do so in the future), this PR uses an additional level of batching to limit the amount of memory consumed.

### CODE REVIEW

File: dashboard/app/controllers/pd/professional_learning_landing_controller.rb [hunk 1]
```diff
@@ -27,8 +27,9 @@ def index
   end
 
   def applications_closed
+    false
```
Comment (megcrenshaw) [inline dashboard/app/controllers/pd/professional_learning_landing_controller.rb]: Could you give more context about this change?

**Review by megcrenshaw — COMMENTED** [review]
Am I reading it right that `create_rollup_tables` will re-populate all status logs every week? Why would there not be duplicate entries in the table?

**Comment by TurnerRiley** [comment]
> Am I reading it right that `create_rollup_tables` will re-populate all status logs every week? Why would there not be duplicate entries in the table?

Overall updates, can confirm that with the new change the cronjob runs smoothly and recreates the same behavior as what Dave listed before. As for this question here, I'm not sure why there are no duplicates but even re-running the cronjob a couple times back to back still listed the same number of entries and the way it's written mirrors that of other tables as well.

File: bin/cron/create_rollup_tables [hunk 2]
```diff
@@ -1,7 +1,10 @@
 #!/usr/bin/env ruby
 
-# This script populates (after truncating) the contained_levels,
-# contained_level_answers, and the level_sources_multi_type tables.
+# This script truncates and then populates the following tables:
+#   contained_levels
+#   contained_level_answers
+#   level_sources_multi_type
+#   pd_applications_status_logs
```
Comment (megcrenshaw) [inline bin/cron/create_rollup_tables:7]: Love this formatting!

**Review by megcrenshaw — APPROVED** [review]
Thank you for taking another look at this and verifying that it's working! This looks great.

File: dashboard/app/controllers/pd/professional_learning_landing_controller.rb (hunk 1 above)
Comment (davidsbailey) [reply dashboard/app/controllers/pd/professional_learning_landing_controller.rb]: thank you Meg, nice catch!

**Comment by davidsbailey** [comment]
> 

the previous contents of the table are removed on each run by this line: https://github.com/code-dot-org/code-dot-org/blob/d33de0c45ab52cc2aa5f04ed5ea425f1aad339e6/bin/cron/create_rollup_tables#L228

**Pull request merged**